src/Halogen/Menu/Component.purs

```diff
@@ -0,0 +1,124 @@
+module Halogen.Menu.Component where
```

Comment:
I also went with `State` over `Model`, as `State` is component specific stuff, `Model` suggests some kind of general data model for a domain, but that is not the case for component state values/types.